File: core/server/project/commonBase/src/main/java/com/home/commonBase/config/base/BaseConfig.java
package com.home.commonBase.config.base;

import com.home.shine.data.BaseData;

/** 配置基类 */
public class BaseConfig extends BaseData
{
	@Override
	protected final void afterRead()
	{
		afterReadConfig();
	}
	
	/** 读完配置 */
	protected void afterReadConfig()
	{
		
	}
	
	/** 更新内容 */
	public void refresh()
	{
		generateRefresh();
	}
	
	/** 生成的刷新 */
	protected void generateRefresh()
	{
	
	}
}

File: core/clientLaya/game/src/commonGame/config/base/BaseConfig.ts
namespace Shine
{
	/** 基础配置 */
    export class BaseConfig extends BaseData
    {
        constructor()
        {
            super();
        }
        
        protected afterRead(): void
        {
            this.afterReadConfig();
        }

        /** 读完单个配置 */
        protected afterReadConfig():void
        {

        }

        /** 更新内容 */
        public refresh():void
        {
            this.generateRefresh();
        }

        /** 生成的刷新 */
        protected generateRefresh():void
        {

        }
    }
}